js/interface.js

```diff
@@ -599,8 +629,124 @@ function activateFind() {
   }
 }
 
-function sortDataSources(key, order) {
-  var toBeOrderedDataSources = dataSources;
+function restoreItem(id, name) {
+  Fliplet.API.request({
+    url: 'v1/data-sources/' + id + '/restore',
+    method: 'POST'
+  }).then(function() {
+    $('.data-source[data-id="' + id + '"]').remove();
+
+    trashSources = trashSources.filter(function(ds) {
+      return ds.id !== id;
+    });
+
+    Fliplet.Modal.alert({
+      title: 'Restore complete',
+      message: '"' + name + '" restored'
+    });
+  }).catch(function(error) {
+    Fliplet.Modal.alert({
+      title: 'Restore failed',
+      message: Fliplet.parseError(error)
+    });
+  });
+  currentDataSourceId = 0;
+}
+
+function removeTrashItem(id, name) {
+  Fliplet.Modal.prompt({
+    title: '<p>Delete data source</p><br/><span>Enter the data source name <code>' + name + '</code> to confirm.</span>',
+    value: null,
+    maxlength: 255,
+    buttons: {
+      confirm: {
+        label: 'Delete data source',
+        className: 'btn-danger'
+      },
+      cancel: {
+        label: 'Cancel',
+        className: 'btn-default'
+      }
+    }
+  }).then(function(result) {
+    if (result === null) {
+      return;
+    }
+
+    if (result === name.toString()) {
+      Fliplet.API.request({
+        url: 'v1/data-sources/deleted/' + id,
+        method: 'DELETE'
+      }).then(function() {
+        // Remove from UI
+        $('.data-source[data-id="' + id + '"]').remove();
+
+        // Remove from trashSources
+        trashSources = trashSources.filter(function(ds) {
+          return ds.id !== id;
+        });
+
+        renderTrashSources(trashSources);
+
+        Fliplet.Modal.alert({
+          title: 'Deletion complete',
+          message: 'Item deleted permanently.'
+        });
+
+        // Return to parent widget if in overlay
+        if (copyData.context === 'overlay') {
+          Fliplet.Studio.emit('close-overlay');
+          return;
+        }
+      }).catch(function(error) {
+        Fliplet.Modal.alert({
+          title: 'Deletion failed',
+          message: Fliplet.parseError(error)
+        });
+      });
+    } else {
+      Fliplet.Modal.alert({
+        title: 'Deletion failed',
+        message: 'Data source name is incorrect'
+      }).then(function() {
+        removeTrashItem(id, name);
+      });
+    }
+    currentDataSourceId = 0;
+  });
+}
+function deleteItem(message, currentDataSourceId) {
+  Fliplet.Modal.confirm({
+    message: message
+  }).then(function(confirmAlert) {
+    if (confirmAlert) {
```

Comment:
again, please use early return instead of indenting all those lines

Reply:
@squallstar fixed